### PR TITLE
Validate homeserver configuration prior to loading the app

### PR DESCRIPTION
**Pull request by turt2live**
See https://github.com/matrix-org/matrix-react-sdk/pull/3001

There are no new commits on this branch. The kind of review for this PR is really just a formal green light to merge this whole thing - the individual parts have already been merged. This PR also serves as a changelog entry for when we release.

Includes:
* https://github.com/vector-im/riot-web/pull/9496
* https://github.com/vector-im/riot-web/pull/9719
* https://github.com/vector-im/riot-web/pull/9721
* https://github.com/vector-im/riot-web/pull/9726

See react-sdk PR at the top of this post for list of fixed issues.

### CODE REVIEW

File: src/vector/index.js [hunk 1]
```diff
@@ -45,6 +45,9 @@ import VectorConferenceHandler from 'matrix-react-sdk/lib/VectorConferenceHandle
 import Promise from 'bluebird';
 import request from 'browser-request';
 import * as languageHandler from 'matrix-react-sdk/lib/languageHandler';
+import {_t, _td, newTranslatableError} from 'matrix-react-sdk/lib/languageHandler';
```
Comment (jryans) [inline src/vector/index.js]: Bump NV Ltd to 2019

File: src/vector/mobile_guide/index.js [hunk 2]
```diff
@@ -6,19 +6,102 @@ function onBackToRiotClick() {
...
     document.getElementById('back_to_riot_button').onclick = onBackToRiotClick;
 
-    const config = await getVectorConfig('..');
-    let hsUrl;
-    if (config && config['default_hs_url']) {
-        hsUrl = config['default_hs_url'];
+    let config = await getVectorConfig('..');
+
+    // We manually parse the config similar to how validateServerConfig works because
+    // calling that function pulls in roughly 4mb of JS we don't use.
```
Comment (jryans) [inline src/vector/mobile_guide/index.js]: One day we'll actually fix our build to make sense... 😭

File: test/app-tests/loading.js [hunk 3]
```diff
@@ -616,11 +626,21 @@ describe('loading:', function() {
...
 
+        // When we switch to the login component, it'll hit the login endpoint
+        // for proof of life and to get flows. We'll only give it one option.
+        httpBackend.when('GET', '/login')
+            .respond(200, {"flows": [{"type": "m.login.password"}]});
+        httpBackend.flush(); // We already would have tried the GET /login request
+
+        // Give the component some time to finish processing the login flows before
+        // continuing.
+        await Promise.delay(100);
```
Comment (jryans) [inline test/app-tests/loading.js]: Hmm, usually adding a timeout like this is setting the test up to fail if it runs on very slow hardware, etc. Is there anything you could do instead, like perhaps wait for some event or element to appear? (It looks like we don't have a good answer to this problem for app tests today, so mostly just trying to get more people thinking about.)

Comment (turt2live) [reply test/app-tests/loading.js]: I'd love to wait for a component, but we don't have a reliable component to watch for changes.

Comment (jryans) [reply test/app-tests/loading.js]: Hmm, so there's no visible change before and after processing the login flow?

Comment (turt2live) [reply test/app-tests/loading.js]: There's a loading icon, but it means far too many things. I suppose we could stick a class on it, but ultimately I'm tempted to just ignore the problem. We only really need 10/100 of those milliseconds, and slow machines are likely to have other problems than a random test.

Comment (jryans) [reply test/app-tests/loading.js]: Well, in an ideal world, test wouldn't have any time dependence like this, but no easy solution for now it seems.

**Comment by turt2live** [comment]
RC cut, let's land this thing 🎉

**Pull request merged**